cms/jinja2/assets/js/hide-date-text-on-non-provisional-releases.js

```diff
@@ -0,0 +1,16 @@
+function toggleDateTextVisibility() {
+  var statusField = document.getElementById('id_status');
+  const isProvisional = statusField.value.toLowerCase() === 'provisional';
+
+  const dateTextInput = document.getElementById('id_release_date_text');
+  const panelWrapper = dateTextInput.closest('.w-panel__wrapper');
+
+  panelWrapper.style.display = isProvisional ? 'block' : 'none';
+}
+
+document.addEventListener('DOMContentLoaded', function () {
+  toggleDateTextVisibility();
+
+  var statusField = document.getElementById('id_status');
+  statusField.addEventListener('change', toggleDateTextVisibility);
```

Comment:
This code gets run in every editor context, so `statusField` can be `null`. You should only add the event listener if `statusField` exists, otherwise you will get errors in the console.

Reply:
I was also wondering if there's maybe a way to access the page details here? If so we could only include the Release pages for this.  Page type is known at this point but is it accessible in JS? 

Or maybe there's a way to use the `insert_editor_js` hook only for given page types? 

Either one would exclude this code sooner, but I'm happy to go with this if there's no mechanism to do it.

Reply:
Afaik conditional inclusion isn't possible using hooks, but I don't think it's a problem to include this in all editor instances, provided that we do the check mentioned.

Reply:
[d5632aa](https://github.com/ONSdigital/dis-wagtail/pull/158/commits/d5632aa9f8ce5f1e73229f8d81c88af5c4a2424c)